src/reducers/reducer.js

```diff
@@ -1072,11 +1082,45 @@ export const onSetWCConnectionFailed = (state, { payload }) => ({
   },
 });
 
+/**
+ * @param {Object} action.payload The network settings emitted in saga
+ * @see updateNetworkSettings
+ */
+export const onNetworkSettingsUpdate = (state) => ({
+  ...state,
+  networkSettingsStatus: NETWORKSETTINGS_STATUS.LOADING,
+});
+
 /**
  * @param {Object} action.payload The network settings emitted in saga
  * @see updateNetworkSettings
  */
 export const onNetworkSettingsUpdateSucess = (state, { payload }) => ({
   ...state,
-  networkSettings: payload
+  networkSettings: payload,
+  networkSettingsStatus: NETWORKSETTINGS_STATUS.LOADING,
```

Comment:
Why do you keep as loading after success?

Reply:
Oh! I believe these actions names need a refactor. This "success" actually means "Ok, you can persist it in the store now". After the "success" action the persist effect will execute, and after it the wallet reloading.

I wrote the refactor necessity in this commit on the PR(2): https://github.com/HathorNetwork/hathor-wallet-mobile/pull/336/commits/095510fe961f22eed8c3b2aa06690515c6e19930 

But now I'm opening an issue to address.

Reply:
Here is the issue to address: 
https://github.com/HathorNetwork/hathor-wallet-mobile/issues/358